ToolBox/Program.cs

```diff
@@ -302,5 +307,71 @@ private static string GetParameterOrDefault(IReadOnlyDictionary<string, object>
 
             return value.ToString();
         }
+
+        /// <summary>
+        /// Loads values from JSON text file
+        /// </summary>
+        private static Dictionary<string, object> LoadOptionsObjectFromConfigurationFile(string path)
+        {
+            var optionsObject = new Dictionary<string, object>();
+            try
+            {
+                // Read the text from json and fill that in a specifal object that
+                // will hold the values corresponding to json file contents
+                var toolBoxOptionsConfig = JsonConvert.DeserializeObject<ToolBoxOptionsConfiguration>(File.ReadAllText(path));
+
+                // iterate over the properties using reflection and check for non empty values to pupulate optionsObject.
+                foreach (var prop in toolBoxOptionsConfig.GetType().GetProperties())
+                {
+                    if (prop.PropertyType == typeof(string))
+                    {
+                        // most of our values are strings, get value, if null or empty then continue; populate otherwise..
+                        var valueAsString = (string) prop.GetValue(toolBoxOptionsConfig);
+                        if (valueAsString.IsNullOrEmpty())
+                            continue;
+
+                        // be cautios of System.InvalidOperationException can happen here if there is no such attribute
+                        var key = GetThePropertyNameFromJsonAttribute(prop);
+                        optionsObject[key] = valueAsString;
+                    }
+                    else if(prop.PropertyType == typeof(string[]))
+                    {
+                        // if we reached here we are most probably working with 'tickers' array
+                        var valueAsStringArray = (string[])prop.GetValue(toolBoxOptionsConfig);
+                        if (valueAsStringArray.IsNullOrEmpty())
+                            continue;
+
+                        var key = GetThePropertyNameFromJsonAttribute(prop);
+
+                        // I fully copy here a piece of code from ApplicationParser.cs that has to do with
+                        // case CommandOptionType.MultipleValue - for the full compatibility!
```

Comment:
The fact that we're _copying_ code from `ApplicationParser` is another reason why this logic belong _inside_ `ApplicationParser`... that and it's parsing options for the application :)